libr/debug/p/debug_native.c

```diff
@@ -1259,6 +1259,52 @@ static int r_debug_desc_native_open (const char *path) {
 	return 0;
 }
 
+static int r_debug_setup_ownership (int fd, RDebug *dbg) {
+	RDebugInfo *info = r_debug_info (dbg, NULL);
+
+	if (!info) {
+		eprintf ("Error while getting debug info.\n");
+		return -1;
+	}
+	fchown (fd, info->uid, info->gid);
+	r_debug_info_free (info);
+  	return 0;
+}
+
+static char *r_debug_setup_name (const char *raw_name, pid_t pid) {
+	return (!*raw_name)?
+		r_str_newf ("core.%u", pid) :
+		r_str_chop (strdup (raw_name));
+}
+
+static bool r_debug_gcore (RDebug *dbg, const char *newcorefile) {
+	RBuffer *buffer = NULL;
+	RBuffer *destination = NULL;
+
+	char *corefile = r_debug_setup_name (newcorefile, dbg->pid);
```

Comment:
This function should return an RBuffer, and this buffer must be saved to disk by RCore.

One of the problems of using r_buf_new is that it will first allocate all in memory, which can be a lot, so i understand the need to have a way to directly write this to disk. 

So i'm writing a new RBuffer constructor. So just change the function signature to this:

`static bool r_debug_gcore (RDebug *dbg, RBuffer *buf) { ... `

this way, the function will be called from core and core will create the new buffer like this: `RBuffer *r_buf_new_file (corefile);` in this buffer, the changes are saved to disk immediately. So you can just r_buf_free (buf); and the file will be created. you can later change the file permissions from the info from r_debug_info. but from what i said, i think it will be better not to chown.